### ISSUE TITLE
zls config memory leak? segfault at detectLeaksInBucket

### ISSUE THREAD

**Issue by guidorice** [zigtools/zls#503]
Using zls 327e66c12b68f7974f81cb11cc618e789dc45a6a and `zig/0.10.0-dev.2523+d9b0c984a` I noticed this crash:

```
$ ./zig-out/bin/zls config
Welcome to the ZLS configuration wizard!
      *
       |\
      /* \
      |  *\
    _/_*___|_    x
      | @ @     /
     @     \   /
      \__-/   /

? Should this configuration be system-wide? (y/n) > n
Found zig executable '/usr/local/bin/zig' in PATH.
? Which code editor do you use? (select one)

  - VSCode
  - Sublime
  - Kate
  - Neovim
  - Vim8
  - Emacs
  - Doom
  - Spacemacs
  - Other

> VSCode
? Do you want to enable snippets? (y/n) > y
? Do you want to enable style warnings? (y/n) > y
? Do you want to enable semantic highlighting? (y/n) > y
? Do you want to enable .* and .? completions? (y/n) > y
Writing config to /Users/alex/Library/Application Support/zls.json ... successful.



To use ZLS in Visual Studio Code, install the 'ZLS for VSCode' extension from 
'https://github.com/zigtools/zls-vscode/releases' or via the extensions menu.
Then, open VSCode's 'settings.json' file, and add:

"zigLanguageClient.path": "[command_or_path_to_zls]"

Thank you for choosing ZLS!
Segmentation fault at address 0x0
/Users/alex/zig/0.10.0-dev.2523+d9b0c984a/files/lib/zig/std/heap/general_purpose_allocator.zig:340:66: 0x10eaa2dcc in std.heap.general_purpose_allocator.GeneralPurposeAllocator((struct std.heap.general_purpose_allocator.Config constant)).detectLeaksInBucket (zls)
```

The stack trace is pretty long, but it mentions:

```
/Users/alex/repos/zig/zls/src/main.zig:1857:31: 0x10ea448f1 in main (zls)
    defer _ = gpa_state.deinit();
```

**Comment by gerred** [comment]
Not going to create a separate issue since it's happening at the same point, on the same commit (327e66c), but the addresses manifest a little differently. I'm on an M1 Max machine running MacOS 12.3. This occurs on zig `v0.10.0-dev.4580+3cb387338`, and I tested a build of Zig with the same commit @guidorice is using with the same stack trace. It still generates out a fresh zls.json file so functionally it's OK, and I haven't run into any other ZLS crashes but watching for that.

Note, `zig build -Drelease-safe` reports a different entry point into the segfault, if anyone else is running into this make sure you remove `zig-out` and `zig-cache` before performing `zig build`. Initially, I couldn't reproduce @guidorice's result until I took those steps.
